packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -2060,33 +2073,56 @@ function intoDeltaImpl(
 	change: FieldChangeMap,
 	nodeChanges: ChangeAtomIdBTree<NodeChangeset>,
 	fieldKinds: ReadonlyMap<FieldKindIdentifier, FieldKindWithEditor>,
-): Map<FieldKey, DeltaFieldChanges> {
+): [Map<FieldKey, DeltaFieldChanges>, DeltaDetachedNodeChanges[], DeltaDetachedNodeRename[]] {
 	const delta: Map<FieldKey, DeltaFieldChanges> = new Map();
+	const global: DeltaDetachedNodeChanges[] = [];
+	const rename: DeltaDetachedNodeRename[] = [];
+
 	for (const [field, fieldChange] of change) {
-		const deltaField = getChangeHandler(fieldKinds, fieldChange.fieldKind).intoDelta(
+		const {
+			local: fieldChanges,
+			global: fieldGlobal,
+			rename: fieldRename,
+		} = getChangeHandler(fieldKinds, fieldChange.fieldKind).intoDelta(
 			fieldChange.change,
-			(childChange): DeltaFieldMap => {
+			(childChange) => {
 				const nodeChange = nodeChangeFromId(nodeChanges, childChange);
-				return deltaFromNodeChange(nodeChange, nodeChanges, fieldKinds);
+				const nodeChangeDelta = deltaFromNodeChange(nodeChange, nodeChanges, fieldKinds);
+				if (nodeChangeDelta !== undefined) {
+					const [nodeFieldChanges, nodeGlobals, nodeRenames] = nodeChangeDelta;
+					if (nodeGlobals.length > 0) {
+						nodeGlobals.forEach((c) => global.push(c));
+					}
+					if (nodeRenames.length > 0) {
+						nodeRenames.forEach((r) => rename.push(r));
+					}
+					return nodeFieldChanges;
+				}
 			},
 		);
-		if (!isEmptyFieldChanges(deltaField)) {
-			delta.set(field, deltaField);
+		if (fieldChanges !== undefined && fieldChanges.length > 0) {
+			delta.set(field, fieldChanges);
+		}
+		if (fieldGlobal !== undefined) {
+			fieldGlobal.forEach((c) => global.push(c));
+		}
+		if (fieldRename !== undefined) {
+			fieldRename.forEach((r) => rename.push(r));
 		}
 	}
-	return delta;
+	return [delta, global, rename];
 }
 
 function deltaFromNodeChange(
 	change: NodeChangeset,
 	nodeChanges: ChangeAtomIdBTree<NodeChangeset>,
 	fieldKinds: ReadonlyMap<FieldKindIdentifier, FieldKindWithEditor>,
-): DeltaFieldMap {
+): [DeltaFieldMap, DeltaDetachedNodeChanges[], DeltaDetachedNodeRename[]] | undefined {
```

Comment:
What is the motivation behind allowing `undefined` to be returned as part of this PR?

Reply:
There isn't really any other than the todo, I'll revert all of it